narwhals/_spark_like/utils.py

```diff
@@ -58,6 +62,45 @@ def native_to_narwhals_dtype(
     return dtypes.Unknown()
 
 
+def narwhals_to_native_dtype(
+    dtype: DType | type[DType], version: Version
+) -> pyspark_types.DataType:
+    from pyspark.sql import types as pyspark_types
+
+    dtypes = import_dtypes_module(version)
+
+    if isinstance_or_issubclass(dtype, dtypes.Float64):
+        return pyspark_types.DoubleType()
+    if isinstance_or_issubclass(dtype, dtypes.Float32):
+        return pyspark_types.FloatType()
+    if isinstance_or_issubclass(dtype, dtypes.Int64):
+        return pyspark_types.LongType()
+    if isinstance_or_issubclass(dtype, dtypes.Int32):
+        return pyspark_types.IntegerType()
+    if isinstance_or_issubclass(dtype, dtypes.Int16):
+        return pyspark_types.ShortType()
+    if isinstance_or_issubclass(dtype, dtypes.Int8):
+        return pyspark_types.ByteType()
+    if isinstance_or_issubclass(dtype, dtypes.String):
+        return pyspark_types.StringType()
+    if isinstance_or_issubclass(dtype, dtypes.Boolean):
+        return pyspark_types.BooleanType()
+    if isinstance_or_issubclass(dtype, dtypes.Datetime):
+        return pyspark_types.TimestampType()
+    if isinstance_or_issubclass(dtype, dtypes.List):  # pragma: no cover
+        msg = "Converting to List dtype is not supported yet"
+        return NotImplementedError(msg)
+    if isinstance_or_issubclass(dtype, dtypes.Struct):  # pragma: no cover
+        msg = "Converting to Struct dtype is not supported yet"
+        return NotImplementedError(msg)
+    if isinstance_or_issubclass(dtype, dtypes.Array):  # pragma: no cover
+        msg = "Converting to Array dtype is not supported yet"
+        return NotImplementedError(msg)
+
+    msg = f"Unknown dtype: {dtype}"  # pragma: no cover
+    raise AssertionError(msg)
```

Comment:
Should we raise a NotImplementedError for unsigned integer as well?
I would prefer that rather than falling into the unknown types

Reply:
Sounds good! it is the same thing we do in duckdb for Categorical types.

I think it may be useful to distinguish between:
- error we raise because we haven't implemented the type yet (-> `NotImplementedError`)
- error we raise because the type is not implemented by the backend (-> e.g.`UnsupportedDTypeError`?)

What do you think? Is it something that could be useful?

Reply:
I like the idea, it would definitly be more clear from a user perspective